### PR TITLE
Plasmidfinder

### PR DESCRIPTION
**Pull request by pimarin**
I update the .shed file to use the github IUC url instead of my own

### CODE REVIEW

File: tools/plasmidfinder/plasmidfinder.xml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    <expand macro="requirements" />
+    <command detect_errors="aggressive"><![CDATA[
+        mkdir output_dir &&
+        mkdir temp_dir &&
+        plasmidfinder.py
+        -i '$input.input_file'
+        -p '$input.database_name.fields.path'
+        -l '$options.min_cov'
+        -t '$options.threshold'
+        #if $input.input_file.ext == 'fasta' or $input.input_file.ext == 'fasta.gz'
```
Comment (bernt-matthias) [inline tools/plasmidfinder/plasmidfinder.xml:18]: You could simplify this by using `$input.input_file.ext.startswith("fasta")`

File: tools/plasmidfinder/plasmidfinder.xml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        -tmp temp_dir
+        #*======================================
+                            LOG file
+        ======================================*#
+        | tee '$log_file'
+        ]]>
+    </command>
+    <inputs>
+        <section name="input" title="Input parameters" expanded="true">
+            <param name="input_file" type="data" format="fasta,fastq,fasta.gz,fastq.gz" label="Choose a fasta or fastq file" help="File to be analyzed"/>
```
Comment (bernt-matthias) [inline tools/plasmidfinder/plasmidfinder.xml:34]: fastqsanger instead of fastq?

File: tools/plasmidfinder/plasmidfinder.xml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        <section name="options" title="Options">
+            <param name="min_cov" type="float" min="0" max="1" value="0.6" label="Minimal coverage" help="Choose a minimum coverage value (default: 0.6)"/>
+            <param name="threshold" type="float" min="0" max="1" value="0.95" label="Minimal identity" help="Choose a minimum identity value (default: 0.95)"/>
+        </section>
+        <section name="output_files" title="Output file selection">
+            <param name="output_selection" type="select" display="checkboxes" multiple="true" label="Output files selection">
+              <option value="data_json">JSON file result</option>
+              <option value="hit_fasta" selected="true">Hits in genome</option>
+              <option value="plasmid_fasta" selected="true">Plasmid hits</option>
+              <option value="result_tsv" selected="true">Plasmid results</option>
```
Comment (bernt-matthias) [inline tools/plasmidfinder/plasmidfinder.xml:50]: The description shown to the user could be clearer. Maybe hits -> sequences, and some better description what kind of results. 

Ideally the description would also match the suffix of the output labels (did not check).

File: tools/plasmidfinder/plasmidfinder.xml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                  <param name="output_selection" value="hit_fasta,plasmid_fasta,result_tsv"/>
+              </section>
+              <output name="hit_file" value="test_3/Hit_in_genome_seq_test3.fsa" ftype="fasta"/>
+              <output name="plasmid_file" value="test_3/Plasmid_seqs_test3.fsa" ftype="fasta"/>
+              <output name="result_file" value="test_3/results_tab_test3.tsv"/>
+        </test>
+    </tests>
+    <help><![CDATA[
+      **What it does**
+      PlasmidFinder characterize plasmid sequences into whole genome sequencing.
```
Comment (bernt-matthias) [inline tools/plasmidfinder/plasmidfinder.xml:130]: into -> in

File: tools/plasmidfinder/plasmidfinder.xml [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      It is based on [plasmidfinder database](https://bitbucket.org/genomicepidemiology/plasmidfinder_db/) with hundreds sequences.
```
Comment (bernt-matthias) [inline tools/plasmidfinder/plasmidfinder.xml:131]: on the

File: tools/plasmidfinder/plasmidfinder.xml [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+      **Input**
+      It can analyse raw data using a k-mer approach based on KMA or a blastn for genome assembly.
+      A database is need obtained from the plasmidfinder database.
```
Comment (bernt-matthias) [inline tools/plasmidfinder/plasmidfinder.xml:134]: please reformulate.

File: tools/plasmidfinder/plasmidfinder.xml [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+      **Options**
+      You can modify the coverage value (% of matching sequence on the total target sequence)
+      You can modify the identity threshold (% of shared nucleotide on the match)
```
Comment (bernt-matthias) [inline tools/plasmidfinder/plasmidfinder.xml:137]: its obvious that the user can change the value. Maybe remove here and put the text which is in parentheses into the parameters help attribute.

File: tools/plasmidfinder/plasmidfinder.xml (hunk 1 above)
Comment (nsoranzo) [reply tools/plasmidfinder/plasmidfinder.xml:18]: I think explicit better than implicit in this case? There may come a `fastafoo` format in the future.

**Comment by pimarin** [comment]
I have isescan file which are inetgrated in this PR

**Pull request closed without merging**